test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -55,10 +56,10 @@ test.describe(
 			// Check the markup of the block is correct.
 			await editor.publishPost();
 			const content = await editor.getEditedPostContent();
+			const navigationMenus = await navBlockUtils.getNavigationMenus();
+			const latestNavigationMenu = navigationMenus[ 0 ];
 			expect( content ).toBe(
-				`<!-- wp:navigation -->
-<!-- wp:page-list /-->
-<!-- /wp:navigation -->`
+				`<!-- wp:navigation {"ref":${ latestNavigationMenu.id }} /-->`
```

Comment:
Nit: We can try [`editor.getBlocks`](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md#the-solution) for slightly more readable output.

```js
await expect.poll( editor.getBlocks ).toMatchObject( [ {
  name: 'core/navigation',
  attributes: { ref: latestNavigationMenu.id },
} ] ); 
```